app/Controllers/MediaController.php

```diff
@@ -188,6 +188,50 @@ public function download(Request $request, Response $response, string $userCode,
         return $this->streamMedia($request, $response, $this->storage, $media, 'attachment');
     }
 
+
+    /**
+     * @param  Request  $request
+     * @param  Response  $response
+     * @param  string  $vanity
+     * @param  string  $id
+     *
+     * @return Response
+     * @throws HttpNotFoundException
+     * @throws HttpBadRequestException
+     */
+    public function createVanity(Request $request, Response $response, int $id): Response
+    {
+        if (!$this->session->get('admin')) {
+            $media = $this->database->query('SELECT * FROM `uploads` WHERE `id` = ? LIMIT 1', $id)->fetch();
+        } else {
+            $media = $this->database->query(
+                'SELECT * FROM `uploads` WHERE `id` = ? AND `user_id` = ? LIMIT 1',
+                [$id, $this->session->get('user_id')]
+            )->fetch();
+        }
+
+        $data = $request->getParsedBody();
+        $vanity = $data['vanity'];
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common processes. Try to leave a link to the process described so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏There is the `param` helper to accoplish this https://github.com/SergiX44/XBackBone/blob/master/app/helpers.php#L279

Reply:
Thanks for pointing this out. I actually did not know about this function from when I looked over your code, but it works well. Fixed :+1: